### PR TITLE
Remove unnecessary Typeable deriving, Solves #1653

### PR DESCRIPTION
**Pull request by juanpaucar**
Solves #1653 

Before submitting your PR, check that you've:

- [x] Bumped the version number
- [ ] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [ ] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddocks for new, public APIs

After submitting your PR:

- [ ] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by juanpaucar** [comment]
@MaxGabriel Could you take a look, please?

File: yesod-auth/yesod-auth.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:            yesod-auth
-version:         1.6.8
+version:         1.6.9
```
Comment (MaxGabriel) [inline yesod-auth/yesod-auth.cabal]: Should this be 1.6.8.1?

File: yesod-core/yesod-core.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:            yesod-core
-version:         1.6.17
+version:         1.6.18
```
Comment (MaxGabriel) [inline yesod-core/yesod-core.cabal]: Should this be 1.6.17.1?

File: yesod/ChangeLog.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 1.6.1
+
+* Remove unnecessary deriving of Typeable
```
Comment (MaxGabriel) [inline yesod/ChangeLog.md]: Should this be 1.6.0.1?

**Comment by MaxGabriel** [comment]
This looks good to me. The only thing I didn’t understand is sometimes you bumped the 4th version number, and sometimes you bumped the third one. I would think the same type of bump applies to all of them?

My guess would be the 4th version number should be bumped, since this should have no impact on library users and is just an internal fix

**Comment by MaxGabriel** [comment]
Also thank you for contributing :)

**Comment by juanpaucar** [comment]
@MaxGabriel Sorry about the confusion on the version bumping. I just corrected the version according to your comments. Please take a look when you have some time

**Comment by MaxGabriel** [comment]
Great work!

**Comment by MaxGabriel** [comment]
Ok, merged and released:

```
 * [new tag]           yesod-1.6.0.1 -> yesod-1.6.0.1
 * [new tag]           yesod-auth-1.6.8.1 -> yesod-auth-1.6.8.1
 * [new tag]           yesod-core-1.6.17.1 -> yesod-core-1.6.17.1
```

I don't have hackage rights to `yesod-static` or `yesod-auth-oauth`. If someone could upload those and push their tags to git, that would be great

**Pull request merged**